flocker/acceptance/testtools.py

```diff
@@ -866,10 +928,12 @@ def create_dataset(test_case, cluster, maximum_size=None, dataset_id=None):
         maximum_size = get_default_volume_size()
     if dataset_id is None:
         dataset_id = uuid4()
-
+    if metadata is None:
+        metadata = {}
+    metadata[u"name"] = metadata.get(u"name", u"my_volume")
```

Comment:
Mutating inputs isn't a great idea, tests might accidently pass-in something that then gets mutated. Make a copy and mutate that.